### ISSUE TITLE
Issue opening modals on Android

### ISSUE THREAD

**Issue by arunmenon1975** [grahammendick/navigation#336]
The router is working great for my app thus far. I have quite a few screens (considering the form factor) and its great to have a standard and predictable routing solutions across the screens.

But i seem to have a problem whenever i attempt to show a `Modal` with my app crashing if the modal has a new navigation handler embedded in it. While my actual implementation is a lot more complicated, i attempted a much simpler example (content taken from #262) in an attempt to reproduce it without any overhead. 

The app does crash with this as well:

```
const modalNavigator = new StateNavigator([
  { key: 'settings' },
  { key: 'details', trackCrumbTrail: true }
]);

modalNavigator.navigate('settings');

const { settings, details } = modalNavigator.states;

settings.renderScene = () => (
  <NavigationContext.Consumer>
    {({ stateNavigator }) => (
      <TouchableHighlight
        style={{ flex: 1 }}
        onPress={() => {
          stateNavigator.navigate('details');
        }}>
          <Text>
          Detail
          </Text>
      </TouchableHighlight>
    )}
  </NavigationContext.Consumer>
);

details.renderScene = () => {
  return (<View style={{ flex: 1 }}>
    <Text>  
      Details 
    </Text>
  </View>)
}

const App: React.FunctionComponent = () => {
  const [showModal, setShowModal] = useState(false);
  const _modalClosed = () => {
    setShowModal(false)
  }
  return (<View style={{ flex: 1 }}>
    <Text onPress={()=>setShowModal(true)}>
      Open Modal
    </Text>
    <View>
    <Modal visible={showModal} onRequestClose={_modalClosed}>
      <NavigationHandler stateNavigator={modalNavigator}>
        <NavigationStack />
      </NavigationHandler>
      
</Modal>
</View>
  </View>)
}

export default App;

```

If i remove the `NavigationHandler` block, it doesn't crash anymore.

I am on `v6.4.1`. 

I have bottom tabs (JS tabs from a single stateNavigator) that then opens top tabs(Navigator tabs). Inside the top tabs i have a few instances where i need a more focussed interface - mostly wizard-like flows. I want to open these scenes 'out' of the top tabs (visually) and present them as standard screens with only perhaps a header i.e no bottom and top tabs. Once the focussed tasks are done (across a few scenes) i intend to return the users back to where they started out with.

There is no straight forward way to do it currently on the router, or at least i haven't found any yet. The alternate option is the Modal approach which was what i was trying until i came across the issue.

**Comment by grahammendick** [comment]
Glad to hear it’s working great for you!

The technique in #262 only works on iOS. It doesn’t work on Android because the React Native Modal doesn’t support Fragments.

We could write our own Modal component that does support Fragments. But I don’t think it’s worth it because we’d still run up against the [React Native Gesture Handler Modal bug](https://github.com/kmagiera/react-native-gesture-handler/issues/139).

A better solution is to create your own subflow using a standard `View` containing a new `NavigationStack`. 

```jsx
const App: React.FunctionComponent = () => {
  const [showModal, setShowModal] = useState(false);
  return (<View style={{flex: 1}}>
    <Text onPress={()=> setShowModal(true)}>
      Open Modal
    </Text>
    {showModal && <View style={{position: 'absolute', top: 20, bottom: 20, left: 20, right: 20}}>
        <NavigationHandler stateNavigator={modalNavigator}>
          <NavigationStack />
        </NavigationHandler>
      </View>}
  </View>)
}
```

**Comment by arunmenon1975** [comment]
ok and yes that does works, thank you. The only issue i have here is more styling related (relative and absolute positioning) considering my current layout and structure. I cannot introduce this view deep inside my app since there are quite a few parent elements above with explicit or implicit relative positioning(from early quick tests) that seems to restrict the boundaries well below the screen bounds. I think what will work is having this view reside at a higher hierarchy and enabling/disabling using some form of global state management or context. Even the stateNavigator object could be passed on similarly so that the view can be reused across the app.

**Comment by grahammendick** [comment]
Moving it up the hierarchy so it can take up the whole screen makes sense. This isn't ideal though and, if RNGH ever fix their Modal bug, then I'll think again about creating a Modal component that works with Fragments on Android

**Comment by arunmenon1975** [comment]
I got it to work on a test approach via redux with the modal view residing higher up and listening to a particular change. This does solve the styling issue for me. The hardware back navigation also works as i navigate across the scenes on the modal stack and this is a big plus.

However when i close the modal(i actually dispatch an action that flips a boolean to effect this ) i get a black screen in place of the earlier screen the modal view was laid over on . The `details.unmountStyle = 'none'` doesn't work as well. 

In my WIP effort i still have my (JS) bottom tabs displayed (haven't moved it further up as yet.  When i tap on any tab everything gets back ok.

**Comment by grahammendick** [comment]
Don't forget to set `primary={false}` on the modal stack.

**Comment by arunmenon1975** [comment]
That was my problem - i had it set on my tabs stack where i had a similar sort of issue but missed adding it here. 

Everything works fine now.

**Comment by grahammendick** [comment]
Hey @arunmenon1975. I've just released `navigation-react-native v6.10.0` with support for  [navigation inside a Modal](https://grahammendick.github.io/navigation/documentation/native/modal.html) on both iOS and Android.
